MekHQ/src/mekhq/gui/dialog/PartsReportDialog.java

```diff
@@ -344,6 +402,65 @@ private void refreshOverviewPartsInUse() {
         column = (PartsInUseTableModel.ButtonColumn) tcm.getColumn(PartsInUseTableModel.COL_BUTTON_GMADD_BULK)
                 .getCellRenderer();
         column.setEnabled(campaign.isGM());
+        topUpGMButton.setEnabled(campaign.isGM());
+    }
+
+    private Set<PartInUse> getPartsInUseFromTable() {
+        Set<PartInUse> partsInUse = new HashSet<PartInUse>();
+        for(int row = 0; row < overviewPartsInUseTable.getRowCount(); row++) {
+            partsInUse.add(overviewPartsModel.getPartInUse(row));
+        }
+        return partsInUse;
+    }
+
+    private void topUp() {
+        campaign.stockUpPartsInUse(getPartsInUseFromTable());
+        refreshOverviewPartsInUse();
+    }
+
+    private void topUpGM() {
+        campaign.stockUpPartsInUseGM(getPartsInUseFromTable());
+        refreshOverviewPartsInUse();
+    }
+
+    private int findTopUpAmount(PartInUse piu) {
+        IAcquisitionWork partToBuy = piu.getPartToBuy();
+        int inventory = piu.getStoreCount() + piu.getTransferCount() + piu.getPlannedCount();
+        int needed = (int)Math.ceil(piu.getRequestedStock()/100.0 * piu.getUseCount());
+        int toBuy = needed-inventory;
+
+        if(piu.getIsBundle() == true) {
+            toBuy = (int)Math.ceil((float)toBuy * piu.getTonnagePerItem() / 5);
+            //special case for ammo, need to track down if there's a way to code this properly
+        }
+
+        if(toBuy > 0) {
+            System.out.println("TPI: " + piu.getTonnagePerItem() + " " + String.format("Inv: %d needed: %d tobuy: %d", inventory, needed, toBuy));
```

Comment:
Let's definitely push this to logger rather than System.out. System.out is fine for temporary logging, _maybe_, but for anything like this we need a record of it when players find issues and send in their logs.

Reply:
these sysstem.outs are for my own internal debugging and I forgot to remove them, they've all been purged from the code